### PR TITLE
preview diff/commit for changes

### PR DESCRIPTION
**Pull request by kanodia-parag**
preview diff

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/99designs/gqlgen
 
-go 1.22.5
+go 1.23
```
Comment (StevenACoffman) [inline go.mod:3]: Hey, can you make sure you don't upgrade this by mistake?

File: codegen/templates/templates.go [hunk 2]
```diff
@@ -274,6 +275,21 @@ func Call(p *types.Func) string {
 	return pkg + p.Name()
 }
 
+func dict(values ...interface{}) (map[string]interface{}, error) {
+    if len(values)%2 != 0 {
+        return nil, fmt.Errorf("invalid dict call: arguments must be key-value pairs")
```
Comment (alexandear) [inline codegen/templates/templates.go:280]:
```suggestion
        return nil, errors.New("invalid dict call: arguments must be key-value pairs")
```

File: codegen/templates/templates.go [hunk 3]
```diff
@@ -274,6 +275,21 @@ func Call(p *types.Func) string {
...
+    }
+    m := make(map[string]interface{}, len(values)/2)
+    for i := 0; i < len(values); i += 2 {
+        key, ok := values[i].(string)
+        if !ok {
+            return nil, fmt.Errorf("dict keys must be strings")
```
Comment (alexandear) [inline codegen/templates/templates.go:286]:
```suggestion
            return nil, errors.New("dict keys must be strings")
```

**Comment by kanodia-parag** [comment]
closing this PR as it was raised with partial changes. Will be raising a new PR in a while with complete changes.

**Pull request closed without merging**